### PR TITLE
feat: move components rendering to tsx

### PR DESCRIPTION
**Pull request by prazdevs**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been reviewed and added / updated if needed (for bug fixes
      /start features)

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

There were still components rendered through `h`.

Fixes: #175 

## What is the new behavior?

- Hygen templates have been updated to generate tsx.
- Existing components now render with tsx.

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## Other information

I did not consider moving `c-modal` to tsx as it may be reworked/worked as talked about in #177

### CODE REVIEW

File: packages/c-close-button/src/c-close-button.tsx [hunk 1]
```diff
@@ -69,21 +63,18 @@ export const CCloseButton = defineComponent({
...
+        <chakra.button
+          as={props.as}
+          type="button"
+          disabled={props.isDisabled}
+          aria-label="Close"
+          __label="icon-button"
+          __css={{ ...baseStyles, ...styles.value }}
+          {...attrs}
+        >
+          {slots.default ? () => getValidChildren(slots) : () => <CCloseIcon />}
```
Comment (TylerAPfledderer) [inline packages/c-close-button/src/c-close-button.tsx:76]: This ternary will not invoke the render, but instead return the string of the render itself. (See the vite playground result)

Suggest using this instead:
```tsx
{() => (slots.default ? getValidChildren(slots) : <CCloseIcon />)}
```

Comment (prazdevs) [reply packages/c-close-button/src/c-close-button.tsx:76]: Woops indeed, thanks for pointing that out, will fix it now :)

**Review by TylerAPfledderer — APPROVED** [review]
LGTM! 😄

**Review by codebender828 — CHANGES_REQUESTED** [review]
Thanks for the PR @prazdevs. appreciate all the help

I think there were a few problems with your PR which I explained in the comments:
- Rendering of some components is changed. see Focus lock component
- Styles of `c-code` component not showing

**Comment by prazdevs** [comment]
Thanks for pointing out these issues, everything is fixed!

File: packages/c-focus-lock/src/c-focus-lock.tsx [hunk 2]
```diff
@@ -160,20 +160,16 @@ export const CFocusLock = defineComponent({
...
-          })
+      const clone = cloneVNode(firstChild, {
+        ref: target,
+        ...attrs,
+        "data-chakra-focus-lock": "",
+      })
+
+      return (
+        <clone ref={target} data-chakra-focus-lock="" {...attrs}>
+          {slots?.defaults?.({ enabled, hasFocus, activate, deactivate })}
```
Comment (codebender828) [inline packages/c-focus-lock/src/c-focus-lock.tsx:171]: This should be `slots?.default?.()`

**Review by codebender828 — CHANGES_REQUESTED** [review]
Looks better. Thanks for the changes @prazdevs. I did find one more big in the slots for the focus lock component.

**Review by zakariasahmane — APPROVED** [review]

**Review by codebender828 — APPROVED** [review]
Thank you @prazdevs ! This is a great pull request. Merging 🌤️

**Pull request merged**